src/formats/notion-api/database-helpers.ts

```diff
@@ -377,10 +374,17 @@ export async function createBaseFile(params: CreateBaseFileParams): Promise<stri
 	const baseFilePath = normalizePath(`${databaseFolderPath}/${databaseName}.base`);
 	
 	// For incremental import: update existing .base file if it exists
-	const existingFile = vault.getAbstractFileByPath(baseFilePath);
-	if (existingFile instanceof TFile) {
+	// Use adapter.exists for reliable check
+	if (await vault.adapter.exists(baseFilePath)) {
 		// Update existing .base file with latest database properties
-		await vault.modify(existingFile, baseContent);
+		const existingFile = vault.getAbstractFileByPath(baseFilePath);
+		if (existingFile instanceof TFile) {
+			await vault.modify(existingFile, baseContent);
+		}
+		else {
+			// File exists but not recognized as TFile, use adapter to write directly
```

Comment:
Is this an error that was being encountered? What steps reproduce the error?

Reply:
I changed this part because, in the `readTypesJson` function of the `src/formats/notion-api/vault-helpers.ts` file, the logic required repeatedly reading and writing the `types.json` file, and I encountered a `File already exist` error. Since I was using the `vault.getAbstractFileByPath` method (similar to the logic here) at the time, I felt that this method seemed to have caching and could not return the latest content in real time after being called. At the same time, I recalled that some users had reported (including a screenshot from @kepano in a Discord chat) encountering similar errors, so I examined all possible situations where this error could occur and changed them to the `valut.adapter.exists` method in similar places.